### DISCUSSION TITLE
如何动态的直接使用@iconify-json图标

### DISCUSSION THREAD

**Discussion by rabbit-lili** [unocss/unocss#3319] [Q&A]
![image](https://github.com/unocss/unocss/assets/104117864/f6d5878b-fc2b-4c9e-8c07-bb4bad32aad2)
这里用到了接口请求获取菜单信息，其中包含icon的名称。222直接使用名称是可以显示的，动态拼接引入就显示不了图标
试了很多方法，还是不大行，有什么好的解决方案嘛？

**Comment by zyyv** [comment]
https://unocss.dev/guide/extracting#safelist

**Comment by lxccc812** [comment]
Return full svg icon name instead of splicing when used
![image](https://github.com/unocss/unocss/assets/110974969/7b5176db-1def-47c1-aa9b-944ba12f84c8)
![image](https://github.com/unocss/unocss/assets/110974969/9067d350-01d3-4e1c-8c5c-80819ebe7849)
